rs/src/contracts/tunnel_environments.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+use crate::contracts::tunnel_service_properties::*;
+use url::Url;
+
+pub struct TunnelEnvironment {
```

Comment:
How is this different from `TunnelServiceProperties`?

Reply:
Ah, missed that. Mostly the same except that it uses a parsed `Url` for the service uri, rather than a String. Could get rid of the duplication.